test/logging/error.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"fmt"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+// StructuredError is an error which can hold arbitrary key-value arguments.
+//
+// TODO(coryrc): The Structured Error is experimental and likely to be removed, but is currently in use in a refactored test.
+type StructuredError interface {
+	error
+	GetValues() []interface{}
+	WithValues(...interface{}) StructuredError
+	DisableValuePrinting()
+	EnableValuePrinting()
+}
+
+type structuredError struct {
+	msg           string
+	keysAndValues []interface{}
+	print         bool
+}
+
+func keysAndValuesToSpewedMap(args ...interface{}) map[string]string {
+	m := make(map[string]string)
```

Comment:
boolean flags are just `--xflag`.

Reply:
I'm confused; this has nothing to do with flags; just key-value structured arguments ala Debugw or go-logr.

Reply:
Hm, can I have just a value without key?